### PR TITLE
ceph: Add makeWrapper to dependencies and nativeBuildInputs

### PR DESCRIPTION
**Pull request by lejonet**
###### Motivation for this change

The ceph-mgr daemon needs to know the location of cephs own-made python modules for some of the modules that you can enable for it.
With wrapProgram, a wrapper is added that sets the proper pythonpath environment variable for the ceph-mgr daemon so its modules can find the ceph python modules.

It also adds cherrypy to the python env used by the python scripts, as its needed for the dashboard module for ceph-mgr.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg test ceph

File: pkgs/tools/filesystems/ceph/generic.nix [hunk 1]
```diff
@@ -159,6 +160,7 @@ stdenv.mkDerivation {
 
   postFixup = ''
     wrapPythonPrograms
+    wrapProgram $out/bin/ceph-mgr --set PYTHONPATH $out/lib/python2.7/site-packages
```
Comment (Mic92) [inline pkgs/tools/filesystems/ceph/generic.nix]:
```diff
-    wrapProgram $out/bin/ceph-mgr --set PYTHONPATH $out/lib/python2.7/site-packages 
+    wrapProgram $out/bin/ceph-mgr --set PYTHONPATH $out/${python2Packages.python.sitePackages}
```

**Comment by fpletz** [comment]
@GrahamcOfBorg test ceph

**Comment by Mic92** [comment]
@fpletz borg will always run into timeouts when compiling ceph.

**Comment by srhb** [comment]
LGTM and fixes 36622

Similar fixes will be needed for several other programs as well, but that can be handled separately. :)

**Comment by lejonet** [comment]
@srhb Yeah, seeing as I don't have all types of hardware and permutations of config, it will have to be handled on a case-by-case basis whenever someone finds something is missing for a ceph component, like how you found https://github.com/NixOS/nixpkgs/issues/36622

**Comment by Mic92** [comment]
backport:
[detached HEAD 38302326ce8] ceph: Add makeWrapper to dependencies and nativeBuildInputs
 Author: Daniel Kuehn <daniel@kuehn.se>
 Date: Wed Mar 21 21:53:02 2018 +0100
 1 file changed, 4 insertions(+), 2 deletions(-)
[detached HEAD 40212de1910] ceph: Change hardcoded site-packages path to utilizing the python2Packages.python.sitePackages attribute instead
 Author: Daniel Kuehn <daniel@kuehn.se>
 Date: Thu Mar 22 21:08:28 2018 +0100
 1 file changed, 1 insertion(+), 1 deletion(-)

**Pull request merged**